### PR TITLE
Enhancement of PR Agent with User Interaction

### PR DESCRIPTION
**Pull request by mrT23**
Type of PR:
**Enhancement**

___
PR Description:
**This PR introduces the ability for the PR Agent to interact with the user by asking questions and processing the user's answers. It also includes changes to the command line interface and the GitHub action runner to support these new features. Additionally, it modifies the configuration settings and the prompts used by the PR Agent.**

___
PR Main Files Walkthrough:
-`pr_agent/agent/pr_agent.py`: The main agent class has been updated to handle new commands for asking questions and processing answers. It also uses the settings to decide whether to ask questions before reviewing a PR.
-`pr_agent/cli.py`: The command line interface has been updated to support the new 'user_answers' command.
-`pr_agent/servers/github_action_runner.py`: The GitHub action runner has been updated to handle the new '/answer' command and to ask questions before reviewing a PR based on the settings.
-`pr_agent/tools/pr_information_from_user.py`: The method for preparing the PR answer has been updated to change the format of the response.
-`pr_agent/tools/pr_reviewer.py`: The PRReviewer class has been updated to handle the new 'is_answer' parameter and to get the user's answers from the PR discussion. It also publishes inline code comments based on the settings.
-`pr_agent/settings/configuration.toml`: The configuration settings have been updated to include a new 'ask_and_reflect' setting for the PR reviewer.
-`pr_agent/settings/pr_information_from_user_prompts.toml`: The prompts for getting information from the user have been updated to change the format of the questions.
-`pr_agent/settings/pr_reviewer_prompts.toml`: The prompts for the PR reviewer have been updated to include insights from the user's answers and to change the format of the questions.

### CODE REVIEW

**Review by BossModus666 — COMMENTED** [review]
Marcelschween99@outlook.com

**Review by BossModus666 — APPROVED** [review]

**Comment by mrT23** [comment]
Preparing pr description...

**Comment by mrT23** [comment]
Preparing pr description...

File: pr_agent/agent/pr_agent.py [hunk 1]
```diff
@@ -2,17 +2,24 @@
...
     def __init__(self):
         pass
 
     async def handle_request(self, pr_url, request) -> bool:
-        if any(cmd in request for cmd in ["/review", "/review_pr"]):
-            await PRReviewer(pr_url).review()
+        if any(cmd in request for cmd in ["/answer"]):
+            await PRReviewer(pr_url, is_answer=True).review()
+        elif any(cmd in request for cmd in ["/review", "/review_pr", "/reflect_and_review"]):
+            if settings.pr_reviewer.ask_and_reflect or any(cmd in request for cmd in ["/reflect_and_review"]):
```
Comment (okotek) [inline pr_agent/agent/pr_agent.py:19]:
```suggestion
            if settings.pr_reviewer.ask_and_reflect or "/reflect_and_review" in request:
```

File: pr_agent/servers/github_action_runner.py [hunk 2]
```diff
@@ -53,8 +54,14 @@ async def run_action():
                 pr_url = event_payload.get("issue", {}).get("pull_request", {}).get("url", None)
                 if pr_url:
                     body = comment_body.strip().lower()
-                    if any(cmd in body for cmd in ["/review", "/review_pr"]):
-                        await PRReviewer(pr_url).review()
+                    if any(cmd in body for cmd in ["/answer"]):
```
Comment (okotek) [inline pr_agent/servers/github_action_runner.py:57]: let me replace that with PRAgent call

File: pr_agent/tools/pr_reviewer.py [hunk 3]
```diff
@@ -9,18 +9,22 @@
 from pr_agent.algo.token_handler import TokenHandler
 from pr_agent.algo.utils import convert_to_markdown, try_fix_json
 from pr_agent.config_loader import settings
-from pr_agent.git_providers import get_git_provider
+from pr_agent.git_providers import get_git_provider, GithubProvider
```
Comment (okotek) [inline pr_agent/tools/pr_reviewer.py:12]:
```suggestion
from pr_agent.git_providers import get_git_provider
```

File: pr_agent/tools/pr_reviewer.py [hunk 4]
```diff
@@ -9,18 +9,22 @@
...
 class PRReviewer:
-    def __init__(self, pr_url: str, cli_mode=False):
+    def __init__(self, pr_url: str, cli_mode=False, is_answer: bool = False):
 
         self.git_provider = get_git_provider()(pr_url)
         self.main_language = get_main_pr_language(
             self.git_provider.get_languages(), self.git_provider.get_files()
         )
+        self.is_answer = is_answer
+        if self.is_answer and type(self.git_provider) != GithubProvider:
```
Comment (okotek) [inline pr_agent/tools/pr_reviewer.py:25]: let me add is_supported in git provider

**Review by okotek — CHANGES_REQUESTED** [review]

File: pr_agent/tools/pr_reviewer.py [hunk 5]
```diff
@@ -118,9 +125,26 @@ def _publish_inline_code_comments(self):
...
+                    content = d['suggestion content']
+
+                    self.git_provider.publish_inline_comment(content, relevant_file, relevant_line_in_file)
+            except KeyError:
+                pass
+
+    def _get_user_answers(self):
+        answer_str = question_str = ""
+        if self.is_answer:
+            discussion_messages = self.git_provider.pr.get_issue_comments()
```
Comment (okotek) [inline pr_agent/tools/pr_reviewer.py]: Let me add that function in the base class

**Review by okotek — APPROVED** [review]

File: pr_agent/tools/pr_reviewer.py [hunk 6]
```diff
@@ -15,12 +15,16 @@
...
 class PRReviewer:
-    def __init__(self, pr_url: str, cli_mode=False):
+    def __init__(self, pr_url: str, cli_mode=False, is_answer: bool = False):
 
         self.git_provider = get_git_provider()(pr_url)
         self.main_language = get_main_pr_language(
             self.git_provider.get_languages(), self.git_provider.get_files()
         )
+        self.is_answer = is_answer
+        if self.is_answer and not self.git_provider.is_supported("get_issue_comments"):
```
[the remark this answers is not in the document]
Comment (okotek) [reply pr_agent/tools/pr_reviewer.py:25]: We should probably do that

**Comment by okotek** [comment]
/answer 
1. These commands are also reflected in the command line interface.
2. It is handled in the Jinja template of the AI prompt.
3. If this setting is set, every review defaults to "reflect and review"

**Pull request merged**